rules/use-ngrx-on-complex-applications/rule.md

```diff
@@ -65,4 +67,140 @@ In this type of scenario, the company details you are editing are not shared wit
+
+**Step 1: Define Actions:**
+
+Actions are defined to be dispatched in your application.
+
+Actions represent events or intents that trigger state changes.
+
+Each action is typically defined as a constant using the `createAction` function
+
+Actions may also include payload data using the props function.
+
+For example:
+
+``` typescript
+import { createAction, props } from '@ngrx/store';
+
+export const loadProducts = createAction('[Product] Load Products');
+export const productsLoaded = createAction('[Product] Products Loaded', props<{ products: Product[] }>());
+
+// Add more actions as needed
+```
+**Step 2: Create Reducers:**
+
+Reducers handle state changes based on dispatched actions.
+
+Reducers are pure functions that take the current state and an action as input and return a new state.
+
+Define the initial state for each state slice and handle state changes in response to specific actions.
+
+Use the `createReducer` function from @ngrx/store to create reducer functions.
+
+For example:
+
+``` typescript
+import { createReducer, on } from '@ngrx/store';
+import { productsLoaded } from '../actions/product.actions';
+
+export interface ProductState {
+  products: Product[];
+}
+
+export const initialState: ProductState = {
+  products: [],
+};
+
+export const productReducer = createReducer(
+  initialState,
+  on(productsLoaded, (state, { products }) => ({ ...state, products })),
+  // Add more reducer functions as needed
+);
+```
+
+**Step 3: Configure the Store:**
+
+Import the necessary NGRX modules, such as StoreModule and StoreDevtoolsModule, into your root module (e.g., app.module.ts).
+
+Provide the root reducer using the reducer property of the StoreModule.forRoot() method.
+
+Optionally, include StoreDevtoolsModule.instrument() to enable the Redux DevTools extension.
+
+For example:
+
+``` typescript
+import { StoreModule } from '@ngrx/store';
+import { StoreDevtoolsModule } from '@ngrx/store-devtools';
+import { rootReducer } from './store/reducers';
+
+@NgModule({
+  imports: [
+    StoreModule.forRoot(rootReducer),
+    StoreDevtoolsModule.instrument(),
+    // Other imports and configurations
+  ],
+  // Other module configurations
+})
+export class AppModule { }
+```
+
+**Step 4: Dispatch Actions and Access State from the UI:**
+
+Import the necessary action creators and selectors into your Angular components or services.
+
+Inject the Store into your components or services.
+
+Dispatch actions using the dispatch() method of the Store to trigger state changes.
+Access the state using selectors and the select() method of the Store.
+
+For example:
+
+``` typescript
+import { loadProducts } from './store/actions/product.actions';
+import { selectProducts } from './store/selectors/product.selectors';
+
+constructor(private store: Store<AppState>) {}
+
+// Dispatching an action
+this.store.dispatch(loadProducts());
+
+// Accessing a State
+this.store.pipe(select(selectProducts)).subscribe((products) => {
```

Comment:
Use `this.store.select(selectProducts)` instead of `pipe(select)`

As per:
- NgRx example https://ngrx.io/guide/store/walkthrough
- https://github.com/ngrx/platform/issues/2710
- .select un-deprecated, then become the recommended approach https://github.com/ngrx/platform/pull/1361#issuecomment-430082234

Reply:
Done - I have replaced `pipe(select)` with `this.store.select(selectProducts)`